lib/edge-runtime/utils.ts

```diff
@@ -23,7 +23,9 @@ export function DBChatMessageToGPT(message: DBChatMessage): ChatGPTMessage {
   if (message.role === "function")
     return {
       role: message.role,
-      content: message.content,
+      // Below conversion to JSON and back to string remove all newlines, indentation etc
+      // which are empty tokens. This can cut tokens by ~half
+      content: JSON.stringify(JSON.parse(message.content)),
```

Comment:
is there a chance message.content could be invalid json?

Reply:
Excellent question!

Reply:
It _shouldn't_ be (we only support `application/json` responses and the mock endpoint outputs a json regardless), but down the line there's no guarantee this stays the same.

Have added a try-catch to stay safe